test/new-e2e/tests/agent-platform/install-script/install_script_test.go

```diff
@@ -74,36 +75,78 @@ func TestInstallScript(t *testing.T) {
 			}
 		}
 
-		t.Run(fmt.Sprintf("test install script on %s %s", osVers, *architecture), func(tt *testing.T) {
+		t.Run(fmt.Sprintf("test install script on %s %s %s", osVers, *architecture, *flavor), func(tt *testing.T) {
 			tt.Parallel()
 			fmt.Printf("Testing %s", osVers)
-			e2e.Run(tt, &installScriptSuite{cwsSupported: cwsSupported}, e2e.EC2VMStackDef(ec2params.WithImageName(platformJSON[*platform][*architecture][osVers], archMapping[*architecture], osMapping[*platform])), params.WithStackName(fmt.Sprintf("install-script-test-%v-%v-%s", os.Getenv("CI_PIPELINE_ID"), osVers, *architecture)))
+			e2e.Run(tt, &installScriptSuite{cwsSupported: cwsSupported}, e2e.EC2VMStackDef(ec2params.WithImageName(platformJSON[*platform][*architecture][osVers], archMapping[*architecture], osMapping[*platform])), params.WithStackName(fmt.Sprintf("install-script-test-%v-%v-%s-%s", os.Getenv("CI_PIPELINE_ID"), osVers, *architecture, *flavor)))
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Don't you need to also pass platform to the stack name ? Good point on adding `CI_PIPELINE_ID` to avoid pulumi lock conflict across pipelines

Reply:
We could however the os version includes the platform name. For example for platform = `debian` the os versions are `debian-10` , `debian-11`, ...
I do no think it would make the stack name more explicit or precise. The platform field is used mostly to have a well structured `platforms.json` file.

Reply:
Ah you might use the ami and remove both osVers, but not big gain. I'm wondering if there is a better way to ensure we don't have stack name conflicts